### PR TITLE
Bootstrap flutter_webrtc + rust + libwebrtc toolchain (#4)

### PR DESCRIPTION
**Pull request by logist322**
## Synopsis

Этот PR добавляет необходимый toolchain для разработки `flutter-webrtc` под WIndows.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains `Draft: ` prefix
    - [x] Name contains issue reference
    - [x] Has `k::` labels applied
    - [x] Has assignee
- [x] Documentation is updated (if required)
- [x] Tests are updated (if required)
- [x] Changes conform code style
- [x] CHANGELOG entry is added (if required)
- [x] FCM (final commit message) is posted
    - [x] and approved
- [x] [Review][l:2] is completed and changes are approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] `Draft: ` prefix is removed
    - [x] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by logist322** [comment]
## FCM
```
Implement toolchain for desktop development under Windows (#5)

- add rust crates
- add Makefile
```

**Review by alexlapa — APPROVED** [review]

**Review by tyranron — CHANGES_REQUESTED** [review]
@logist322 one thing should be done more wisely.

File: crates/libwebrtc-sys/build.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use std::{env, path::PathBuf, process::Command};
+
+fn main() {
+    let path = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    #[rustfmt::skip]
+    let libwebrtc_url = "https://github.com/instrumentisto/libwebrtc-bin/releases/download/97.4692.0.0-r0";
```
Comment (alexlapa) [inline crates/libwebrtc-sys/build.rs:6]: @logist322 ,

> This should be implementent in build.rs of libwertc-sys crate and the link controlled via env var.

> and the link controlled via env var

> env var

File: crates/native/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+target/
```
Comment (alexlapa) [inline crates/native/.gitignore:1]: @logist322 ,

Newline добавьте в конец. По хорошему, IDE'шка должна сама это уметь.

@tyranron ,

Вы этот `.gitignore` выкосили, что логично, так как крейт в вокрспейсе, и `/target` лежит не тут. Но тут немного другая история, мы в `build.rs` вот такое делаем:

```rs
    cbindgen::generate_with_config(&crate_dir, config)
        .unwrap()
        .write_to_file(format!("target/{}.hpp", package_name));
```

Ложить хедеры в корневой `target` не получается, так как переменная `CARGO_BUILD_TARGET_DIR` не прокидывается, по неизвестной причине, а хардкодить `../../target` не хочется, но можем, если Вы за такое решение.

**Review by alexlapa — CHANGES_REQUESTED** [review]

File: crates/native/.gitignore (hunk 2 above)
Comment (tyranron) [reply crates/native/.gitignore:1]: @tyranron it's better to investigate what's the problem with `CARGO_BUILD_TARGET_DIR`. It should be working OK.

Comment (alexlapa) [reply crates/native/.gitignore]: @tyranron ,

> Make cargo export CARGO_TARGET_DIR to build.rs scripts

> Build scripts are intended to have a very scoped output (just in their OUT_DIR), and they're not currently really intended to be used as installation scripts for other sorts of files. In that sense this change was never intended for build scripts.

https://github.com/rust-lang/cargo/pull/7325#issuecomment-528365640

А `OUT_DIR` это вот: `/flutter-webrtc/target/debug/build/flutter-webrtc-native-621c22406cba9de9/out`. Бахаю `../../target`.

**Review by alexlapa — APPROVED** [review]

**Review by tyranron — APPROVED** [review]

**Comment by tyranron** [comment]
@logist322 don't use `review` label on GitHub. Asking regular GitHub review is enough.

**Pull request merged**